src/bin/cargo.rs

```diff
@@ -166,9 +167,41 @@ fn execute(flags: Flags, config: &Config) -> CliResult<Option<()>> {
         // For all other invocations, we're of the form `cargo foo args...`. We
         // use the exact environment arguments to preserve tokens like `--` for
         // example.
-        _ => env::args().collect(),
+        _ => {
+            let mut default_alias = HashMap::new();
+            default_alias.insert("b", "build".to_string());
+            default_alias.insert("t", "test".to_string());
+            default_alias.insert("r", "run".to_string());
+            let mut args: Vec<String> = env::args().collect();
+            let command = args[1].clone();
+            if let Some(new_command) = default_alias.get(&command[..]){
+                args[1] = new_command.clone();
+            }
+            args
+        }
     };
 
+    match config.get_list(&format!("alias.{}", args[1])){
```

Comment:
Ah I'd continue to just split on whitespace as it's what Cargo does elsewhere (rather than writing a quoting parser)